### PR TITLE
[#117] Add network security config to support Proxyman

### PR DESCRIPTION
**Pull request by Tuubz**
https://github.com/nimblehq/android-templates/issues/117

## What happened 👀

To use Proxyman, we need to add network security config files to whitelist the domain of our API

I added this for both `RxJavaTemplate` & `CoroutineTemplate`

Read more: [slides](https://docs.google.com/presentation/d/17QIg1Ru5w2m2MkoSb-NTrkl7RVkhdC_DoMIZsGommk8/edit?usp=sharing)
 
## Insight 📝

- Add `network_security_config.xml` for main
- Add `network_security_config.xml` for staging

This is because we should not allow any proxy configurations in production environment
  
## Proof Of Work 📹

https://user-images.githubusercontent.com/18277915/130769062-5352efda-bedc-4297-b739-280e87d899da.mov

### CODE REVIEW

File: CoroutineTemplate/app/src/staging/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    </debug-overrides>
+
+    <base-config cleartextTrafficPermitted="true">
+        <trust-anchors>
+            <certificates src="system" />
+        </trust-anchors>
+    </base-config>
+
+    <domain-config>
+        <domain includeSubdomains="true">www.jsonplaceholder.typicode.com</domain>
```
Comment (hoangnguyen92dn) [inline CoroutineTemplate/app/src/staging/res/xml/network_security_config.xml:16]: 🤔  So when we update the base URL then we have to update the domain as well. For now, I think we could add the comment to notice that replacing `includeSubdomains` to match the base URL. What do you think?

File: CoroutineTemplate/app/src/main/res/xml/network_security_config.xml [hunk 2]
```diff
@@ -0,0 +1 @@
+<network-security-config />
```
Comment (sleepylee) [inline CoroutineTemplate/app/src/main/res/xml/network_security_config.xml:1]: please add eof line break

File: CoroutineTemplate/app/src/staging/res/xml/network_security_config.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<network-security-config>
+    <debug-overrides>
+        <trust-anchors>
+            <certificates src="user" />
+            <certificates src="system" />
+        </trust-anchors>
+    </debug-overrides>
+
+    <base-config cleartextTrafficPermitted="true">
```
Comment (sleepylee) [inline CoroutineTemplate/app/src/staging/res/xml/network_security_config.xml:9]: I think we don't need this tho? 🤔  or do we need to grant the permission to transmit via `http` somewhere?

Comment (Tuubz) [reply CoroutineTemplate/app/src/staging/res/xml/network_security_config.xml:9]: Aah, I see!

We can remove this indeed, if we don't want to support `http` 🙏

**Comment by Tuubz** [comment]
Updated 🚀

**Review by sleepylee — APPROVED** [review]

**Review by lydiasama — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]

**Comment by luongvo** [comment]
LGTM ✅

**Pull request merged**